evaluator/autoware_trajectory_evaluator/src/trajectory_evaluator_node.cpp

```diff
@@ -0,0 +1,234 @@
+        const auto & closest_point = trajectory.trajectory_points[closest_index];
+        double time_at_pose = (current_time - trajectory.time_stamp).seconds();
+        double time_error = std::abs(time_at_pose - closest_point.time_from_start.seconds());
+
+        time_errors_.push_back(TimeErrorData{
+          current_time,                                  // stamp
+          closest_index,                                 // trajectory_index
+          closest_point.trajectory_point.pose.position,  // position
+          closest_point.time_from_start.seconds(),       // expected_time
+          time_at_pose,                                  // actual_time
+          time_error                                     // time_error
+        });
+
+        metrics_msg_.stamp = current_time;
+
+        MetricMsg expected_time_metric;
+        expected_time_metric.name = "trajectory_metrics/expected_time";
+        expected_time_metric.value = closest_point.time_from_start.seconds();
+        metrics_msg_.metric_array.push_back(expected_time_metric);
+
+        MetricMsg actual_time_metric;
+        actual_time_metric.name = "trajectory_metrics/actual_time";
+        actual_time_metric.value = time_at_pose;
+        metrics_msg_.metric_array.push_back(actual_time_metric);
+
+        MetricMsg time_error_metric;
+        time_error_metric.name = "trajectory_metrics/time_error";
+        time_error_metric.value = time_error;
+        metrics_msg_.metric_array.push_back(time_error_metric);
+
+        MetricMsg closest_traj_x;
+        closest_traj_x.name = "trajectory_metrics/closest_traj_x";
+        closest_traj_x.value = closest_point.trajectory_point.pose.position.x;
+        metrics_msg_.metric_array.push_back(closest_traj_x);
+
+        MetricMsg closest_traj_y;
+        closest_traj_y.name = "trajectory_metrics/closest_traj_y";
+        closest_traj_y.value = closest_point.trajectory_point.pose.position.y;
+        metrics_msg_.metric_array.push_back(closest_traj_y);
+
+        MetricMsg closest_traj_z;
+        closest_traj_z.name = "trajectory_metrics/closest_traj_z";
+        closest_traj_z.value = closest_point.trajectory_point.pose.position.z;
+        metrics_msg_.metric_array.push_back(closest_traj_z);
+
+        MetricMsg current_pose_x;
+        current_pose_x.name = "trajectory_metrics/current_pose_x";
+        current_pose_x.value = pose.position.x;
+        metrics_msg_.metric_array.push_back(closest_traj_x);
+
+        MetricMsg current_pose_y;
+        current_pose_y.name = "trajectory_metrics/current_pose_y";
+        current_pose_y.value = pose.position.y;
+        metrics_msg_.metric_array.push_back(closest_traj_y);
+
+        MetricMsg current_pose_z;
+        current_pose_z.name = "trajectory_metrics/current_pose_z";
+        current_pose_z.value = pose.position.z;
+        metrics_msg_.metric_array.push_back(current_pose_z);
+
+        metrics_pub_->publish(metrics_msg_);
+
+        metrics_pub_->publish(metrics_msg_);
```

Comment:
You are publishing twice, and for each stored trajectory. 
This is too much so please only publish the metrics message once per iteration.